diff -u -r boost_1_71_0.orig/boost/asio/detail/config.hpp boost_1_71_0/boost/asio/detail/config.hpp
--- boost_1_71_0.orig/boost/asio/detail/config.hpp	2019-12-05 16:15:15.632159315 +0100
+++ boost_1_71_0/boost/asio/detail/config.hpp	2019-12-05 16:29:16.818182892 +0100
@@ -825,7 +825,11 @@
 #    if (_LIBCPP_VERSION < 7000)
 #     if (__cplusplus >= 201402)
 #      if __has_include(<experimental/string_view>)
-#       define BOOST_ASIO_HAS_STD_EXPERIMENTAL_STRING_VIEW 1
+#       if __clang_major__ >= 7
+#        undef BOOST_ASIO_HAS_STD_EXPERIMENTAL_STRING_VIEW
+#       else
+#        define BOOST_ASIO_HAS_STD_EXPERIMENTAL_STRING_VIEW 1
+#       endif // __clang_major__ >= 7
 #      endif // __has_include(<experimental/string_view>)
 #     endif // (__cplusplus >= 201402)
 #    endif // (_LIBCPP_VERSION < 7000)
diff -u -r boost_1_71_0.orig/libs/filesystem/src/operations.cpp boost_1_71_0/libs/filesystem/src/operations.cpp
--- boost_1_71_0.orig/libs/filesystem/src/operations.cpp	2019-12-05 16:15:14.360150283 +0100
+++ boost_1_71_0/libs/filesystem/src/operations.cpp	2019-12-05 16:27:07.693254039 +0100
@@ -210,6 +210,21 @@

 # if defined(BOOST_POSIX_API)
 
+#  if defined(__ANDROID__)
+#  define truncate libboost_truncate_wrapper
+// truncate() is present in Android libc only starting from ABI 21, so here's a simple wrapper
+static int libboost_truncate_wrapper(const char *path, off_t length)
+{
+  int fd = open(path, O_WRONLY);
+  if (fd == -1) {
+    return -1;
+  }
+  int status = ftruncate(fd, length);
+  close(fd);
+  return status;
+}
+#  endif
+
 #   define BOOST_SET_CURRENT_DIRECTORY(P)(::chdir(P)== 0)
 #   define BOOST_CREATE_HARD_LINK(F,T)(::link(T, F)== 0)
 #   define BOOST_CREATE_SYMBOLIC_LINK(F,T,Flag)(::symlink(T, F)== 0)
